app/forms/proceedings/client_involvement_type_form.rb

```diff
@@ -4,21 +4,25 @@ class ClientInvolvementTypeForm < BaseForm
 
     attr_accessor :client_involvement_type_ccms_code
 
-    validates :client_involvement_type_ccms_code, presence: true
+    before_validation :assign_client_involvement_type_description,
+                      if: -> { client_involvement_type_ccms_code.present? }
```

Comment:
The change to a `before_validation` matched my mental model of what was happening a bit more clearly —so I made the change.

Let's have a discussion about it though. I'm happy to continue overriding `#save` if this change isn't helpful.

Reply:
There is also a slightly nuanced change in behaviour that means it could be possible to save the `ccms_code` without the `description` if validations were overridden.

That isn't possible though with the current implementation of `BaseForm`.